### PR TITLE
Add CI for rocm

### PR DESCRIPTION
**Pull request by Binyang2014**
Add CI for rocm

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -36,12 +38,25 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 
 cd ${SCRIPT_DIR}/..
 
-docker build -t ${GHCR}:base-${TARGET} \
+docker build -t ${GHCR}-common:base-${TARGET} \
```
Comment (chhwang) [inline docker/build.sh:41]: @Binyang2014 Why is this needed?

File: docker/build.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ baseImageTable=(
     ["cuda12.1"]="nvidia/cuda:12.1.1-devel-ubuntu20.04"
     ["cuda12.2"]="nvidia/cuda:12.2.2-devel-ubuntu20.04"
     ["cuda12.3"]="nvidia/cuda:12.3.2-devel-ubuntu20.04"
+    ["rocm6.2"]="rocm/rocm-terminal:6.2"
```
Comment (chhwang) [inline docker/build.sh:11]: @Binyang2014 How about using another image that already has RCCL installed?

Comment (Binyang2014) [reply docker/build.sh:11]: AMD doesn't have much document about supported images. Only can find rocm/rocm-terminal, rocm/pytorch and rocm/tensorflow in the official docs, (https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/docker-image-support-matrix.html) Let me ask AMD teams if they have dev image for this.

File: docker/build.sh (hunk 1 above)
Comment (Binyang2014) [reply docker/build.sh:41]: Due to need to install rccl, will check AMD guys for image supported.

**Review by chhwang — APPROVED** [review]
